ONSdigital/eq-questionnaire-runner
# Adopt semver for releases and introduce release labels

### What is the context of this PR?
Adds autogenerated release notes functionality and `bug_fix_*` branch support for pull requests.

[Auto-generated release labels based on PR label](https://github.com/petechd/eq-questionnaire-runner/releases/tag/v5.1.1)

### How to review 
1. Fork runner
2. Create runner release pipeline (use [this branch](https://github.com/ONSdigital/eq-pipelines/tree/adopt-semver)), use your own docker repository and point it to fork github resource, similar to:
```
  - name: eq-questionnaire-runner-release
    type: github-release-latest
    source:
      owner: petechd
      repository: eq-questionnaire-runner
      tag_filter: v*
      order_by: time
      access_token: ((eq-github.access_token))
```
3. Create a new PR and attach one of the labels (RHS):
```
    - Breaking Changes
    - New Features
    - Enhancements
    - Bug Fix
    - Documentation
```
4. Release new version, starting with something else than "v3" (autogenerate release notes and check if the release label was generated)
5. Check if the image was added to your GAR and/or GCP after new release triggers your pipeline

### Checklist
